overlord/snapstate/conflict.go

```diff
@@ -170,7 +170,7 @@ func CheckChangeConflictMany(st *state.State, instanceNames []string, ignoreChan
 
 	for _, task := range st.Tasks() {
 		chg := task.Change()
-		if chg == nil || chg.Status().Ready() {
+		if chg == nil || (chg.Status().Ready() && task.IsClean()) {
```

Comment:
This one needs the other change in taskrunner to run the cleanup early, otherwise it may trigger conflicts even after the user facing op has finished.